core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -538,13 +541,22 @@ class ReplicaManager(val config: KafkaConfig,
           val initialLogEndOffset = localReplica.logEndOffset
           val logReadInfo = localReplica.log match {
             case Some(log) =>
-              val adjustedFetchSize = if (Topic.isInternal(topic) && !readOnlyCommitted) Math.max(fetchSize, log.config.maxMessageSize) else fetchSize
+              val adjustedFetchSize =
+                if (Topic.isInternal(topic) && !readOnlyCommitted)
```

Comment:
If we always return the first message independent of its size, there is no need to specialize for the internal topic any more. Could we remove this check?